pkg/pillar/cmd/identitymgr/identitymgr.go

```diff
@@ -44,42 +42,45 @@ var Version = "No version specified"
 
 // Information for handleCreate/Modify/Delete
 type identityContext struct {
-	subEIDConfig    pubsub.Subscription
-	pubEIDStatus    pubsub.Publication
-	subGlobalConfig pubsub.Subscription
-	GCInitialized   bool
+	agentBaseContext agentbase.Context
+	subEIDConfig     pubsub.Subscription
+	pubEIDStatus     pubsub.Publication
+	subGlobalConfig  pubsub.Subscription
+	GCInitialized    bool
 }
 
 var debug = false
 var debugOverride bool // From command line arg
+func newIdentityContext() identityContext {
+	ctx := identityContext{}
 
-func Run(ps *pubsub.PubSub) {
-	versionPtr := flag.Bool("v", false, "Version")
-	debugPtr := flag.Bool("d", false, "Debug flag")
-	flag.Parse()
-	debug = *debugPtr
+	ctx.agentBaseContext.ErrorTime = errorTime
+	ctx.agentBaseContext.AgentName = agentName
+	ctx.agentBaseContext.WarningTime = warningTime
+	ctx.agentBaseContext.NeedWatchdog = true
+	ctx.agentBaseContext.CheckAndCreatePidFile = true
+
+	return ctx
+}
+
+func (ctxPtr *identityContext) AgentBaseContext() *agentbase.Context {
+	return &ctxPtr.agentBaseContext
+}
+
+func (ctxPtr *identityContext) AddAgentSpecificCLIFlags() {
```

Comment:
I am wondering if we should make these two function pointers in AgentBaseContext ? 
Most agent won't need these. We are forcing agents that don't need these to define these dummy functions.
Instead, if we make them function pointers, only agents that have Specific CLI would set that function.
And agent base can check if the function is registered and invoke it if defined.

Thoughts??

Reply:
I don't see how it could be a bad thing. I'll update it to reflect this.